### DISCUSSION TITLE
Confusing list settings

### DISCUSSION THREAD

**Discussion by thany** [MalloyDelacroix/DownloaderForReddit#267] [Q&A] [answered]
The `Lock settings` feature is totally confusing me. It says, in my own words, a list's settings will propagate to reddit objects unless they have their settings locked.

Okay, questions about this, and more 🤨

1. What is a "Reddit object"?
2. What mechanism causes settings to get locked?
3. How does this relate to the `Lock settings` checkbox in the master lists, and the lists I created?
4. What is a "self post"?
5. What is the difference between extracting it, and downloading it?
6. How do I make it download to include the name of each subreddit in the path structure? (when I place `%[subreddit_name]` in there, it inserts the name of the list, e.g. `Master User List`)
7. About comment download settings: if I set all 3 dropdowns to `DO NOT DOWNLOAD` (why the caps?) surely all fields after those do not matter, right? But then why are they still enabled, i.e. what remains their purpose?
8. `Cascade changes` - what changes will cascade? And cascade into where?
9. In general, the `Master User List` and `Master Subreddit List` - They are not selectable in the main window. What is their purpose?

I'm also mystified about the database view and its related settings. Quick filters, default filters, but why is there a database view in the first place? What is its purpose exactly? I just don't understand why I would drill down into subreddits and posts using that kind of view, which, even if it did make sense to me to have it around, is still a confusing piece of UI to work with. Focusing on something works really weird. It seems like you never should be focusing on anything other than posts, otherwise you don't get to see data that's actually there. Really really confusing, maybe until I understand its actual purpose.

You can see I find the new version quite a bit more confusing than the previous major release. I consider myself quite a technical person, but even then there seems to be a lot of jargon I have no way of being familiar with, and a lot of quite confusing UI without much explanation or documentation.

**Comment by MalloyDelacroix** [comment, marked as answer]
The lock settings checkbox on each user/subreddit's settings page makes it so that when the master settings (available in the "Download Defaults" section of the main settings dialog) for the list that the user/subreddit is contained in are changed, the master settings changes do not propagate to the user/subreddit.

If you hover your mouse over some of the settings text, they have tool-tips that provide a little more insight into setting.

1.) A "reddit object" is a user or subreddit.  It's just a shorthand that is used in the code, as from a code standpoint they are basically identical.  I didn't intend to put that terminology into the UI, but I guess it slipped in.

2.) When a list's settings are changed in the main settings dialog, every user/subreddit in that list is iterated over and synced with the lists settings.  Unless they have the locked settings flag.

3.) If the locked settings checkbox is checked on the lists settings, any user/subreddit added to that list will also have the locked settings flag.  So when changes are made to the list's settings after a user/subreddit is added, those changes will not propagate to existing users/subreddits with this flag set.

4.)  A "self post" is reddit's terminology for a post that is a text only post.  They do not link directly to any images, videos, or websites.  Although the text of the post may contain links.

5.)  Extracting is the process of getting the post data from reddit's API, then following the link in the post and pulling the data from the contents container page (ie: imgur, gyfcat, etc.).  This information pulled from reddit about the post is stored in a database table, and the information pulled from the container page is stored on its own table.  Downloading is the process of downloading the actual files from the container site, or a text file of a self-post's text to the users computer.  This does not store new information in the database, but instead on the users hard-drive directly.

It is possible to extract posts from reddit, but not to download them.  This is intended to use cases where the user only needs the data and not a copy of the file or text on their hard-drive.

6.) That is how you include the subreddit's name in the path structure.  I assume when you say it only adds the name of the list, you are referring to the example path shown in this section when using the "Download Defaults" portion of the main settings dialog.  This appears to be a bug in the example text.  It should be correct when downloading.  I will fix this issue.

7.)  They are in all caps because they are enum values with the underscore removed.  I agree, the rest of the fields could be disabled if these settings are set to do not download.  However, adjusting the rest of the download comments settings do still adjust the settings associated with the user/subreddit.  They will not be acted upon if all three flags are set to do not download, but were you to change one of the flags, the settings changed while it was off would still have effect.

8.)  When adjusting a lists settings in the main settings dialog: if you have the cascade changes checkbox checked, all the changes made to that list will propagate down to the users/subreddits that are already in that list.  If the checkbox is not checked, the settings changed will only be set for users/subreddits that are added to the list after changing these settings.

9.) The master user and master subreddit lists are the template for new lists that are created.  When a list is created it must have default settings.  The default settings for new lists are synced with the master user and master subreddit lists.  They are abstract templates, which is why they are not available in the main UI's lists.

**The database view dialog:**  

I completely agree.  It is very confusing.  It's a vision I had for being able to view posts, content, and comments more easily than browsing through folders when there is a lot of content to see spread across many folders.  It also provides some tools for database management (deleting items stored in the database).  A user manual is desperately needed for the app, and this dialog is so complex that it is impractical to use without a manual or lots of trial and error. 

I will try to briefly explain its workings:

The radio selectors at the top indicate which of the five sections is the focal section.  The checkboxes  indicate which sections are visible.  When you click on an item in the list which has the focus, all other categories are changed based on that selection.  

For instance: if you have Reddit Objects selected as the focus and you change the reddit object that is selected:
 * The download sessions section will display only sessions in which content from that user/subreddit was downloaded.
 * The posts session will display only posts made by that user or to that subreddit.
 * The content will display only content posted by that user or to that subreddit.
 * The comments will be only comments made by that user or to that subreddit.

From there the sections cascade from left to right.  So if you then select one of the posts made by that user or to that subreddit, only content and comments extracted from that post will be displayed.

If you have reddit objects set as the focal section, the cascade will still start from download sessions.  So if you change the download session, only posts that were downloaded during that session will be displayed, and so on.

If you choose not to view any of the sections, the cascade is formed without that section.  So, if again you have reddit objects as your focal section, and you have posts hidden:  changing the reddit object will cascade the content section to only content extracted from posts made by that user/to that subreddit, but all content will be displayed and will not be dependent on selecting a post. 

The best use of this dialog for the way that I imagine most people use the app is to use it to see new content.  That is: set the focal object to download sessions.  From there, when you run a download you can see only the content that was downloaded during that session.  Selecting a download session when it is the focus will show you all of the user/subreddits that were downloaded during that run, which posts were extracted, and which content was downloaded.  Each filtered by the section before it, or lumped together by hiding the previous section.  Selecting "Download Sessions" from the "Database" drop-down menu on the main UI sets this up for you automatically.

I'm sure this is still confusing, and I intend to make a manual that explains all of this in more detail, but hopefully it helped and was not just the ramblings of a mad man.
